### PR TITLE
tbi(ui5-test-writer): update AllJourneys.gen.json with generated journeys in existing projects

### PR DESCRIPTION
**Pull request by docirl**
## Description

https://github.com/SAP/open-ux-tools/issues/4969

This change ensures that when OPA test files are generated for an **existing** Fiori elements OData V4 application, the newly generated journey file names are automatically registered in the `AllJourneys.gen.json` file — a step that was previously missing for the existing-app flow.

### What changed

**`fiori-elements-opa-writer.ts`**
- Added a call to the new `addJourneysToAllJourneysFile()` function inside `generateOPAFilesForExistingApp()`, so generated journey names are persisted to `AllJourneys.gen.json` after journey files are written.
- Implemented and exported `addJourneysToAllJourneysFile(generatedJourneys, writeContext)`:
  - Reads the existing `AllJourneys.gen.json` from `<testOutDirPath>/integration/`.
  - Skips execution if the file doesn't exist or no journeys were generated.
  - Appends new entries in the format `<name>Journey.gen` while preventing duplicates.
  - Writes the updated array back to the file.

**`addJourneysToAllJourneysFile.test.ts`** _(new file)_
- Unit tests covering all key scenarios:
  - No-op when `generatedJourneys` is empty.
  - No-op (and no file creation) when `AllJourneys.gen.json` doesn't exist.
  - Correctly appends new journey entries.
  - Skips duplicate entries already present in the file.
  - Handles an empty array or `null` JSON file content gracefully.

## Type of change
- [ ] Bug (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a new feature)
- [ ] Breaking change (Bug or New feature that would cause existing functionality/consumers to not work as expected)
- [ ] Non-Breaking chores (Changes to tools, libraries, build process, documentation, etc)
- [ ] None of the above (Reviewers might ask for more clarification)

## How have you tested?
New unit tests added in `addJourneysToAllJourneysFile.test.ts` covering all edge cases using an in-memory `mem-fs` editor.

## Checklist:
- [ ] The code conforms to the [general development principles](https://github.com/SAP/open-ux-tools/blob/main/docs/Guidelines.md)
- [ ] Supplied as many details as possible on this change
- [ ] The code is easy to read and maintainable by others
- [ ] Corresponding changes to the documentation has been done
- [ ] Already existing and new unit tests pass locally

- [ ] 🔄 Regenerate and Update Summary

<details>
<summary>PR Bot Information</summary>

**Version:** `1.28.2`

- Output Template: Repository PR Template
- Summary Prompt: [Default Prompt](https://github.tools.sap/Code-Change-Intelligence/pr-bot/blob/main/src/services/llm/prompts/summary_instructions_prompt.md)
- Event Trigger: `pull_request.opened`
- Correlation ID: `d1bbc620-8509-11f1-940b-17d3e8c76f48`
- LLM: `anthropic--claude-4.6-sonnet`
- File Content Strategy: Full file content
</details>

### CODE REVIEW

File: packages/ui5-test-writer/src/fiori-elements-opa-writer.ts [hunk 1]
```diff
@@ -888,3 +889,31 @@ function writePageObject(
...
+    }
+    const { testOutDirPath, editor } = writeContext;
+    const allJourneysPath = join(testOutDirPath, 'integration', 'AllJourneys.gen.json');
+    if (!editor.exists(allJourneysPath)) {
+        return;
+    }
+    const existing: string[] = (editor.readJSON(allJourneysPath) as string[]) ?? [];
+    const existingSet = new Set(existing);
+    for (const journey of generatedJourneys) {
+        const fileName = `${journey}Journey.gen`;
```
[the remark this answers is not in the document]
Comment (docirl) [reply packages/ui5-test-writer/src/fiori-elements-opa-writer.ts:915]: file extensions are not needed here

[the remark this answers is not in the document]
Comment (heimwege) [reply packages/ui5-test-writer/src/fiori-elements-opa-writer.ts:915]: Now he's confused because of the ESM imports 🐱

File: packages/ui5-test-writer/src/fiori-elements-opa-writer.ts [hunk 2]
```diff
@@ -888,3 +889,31 @@ function writePageObject(
...
+export function addJourneysToAllJourneysFile(generatedJourneys: string[], writeContext: WriteContext): void {
+    if (generatedJourneys.length === 0) {
+        return;
+    }
+    const { testOutDirPath, editor } = writeContext;
+    const allJourneysPath = join(testOutDirPath, 'integration', 'AllJourneys.gen.json');
+    if (!editor.exists(allJourneysPath)) {
+        return;
+    }
+    const existing: string[] = (editor.readJSON(allJourneysPath) as string[]) ?? [];
```
Comment (heimwege) [inline packages/ui5-test-writer/src/fiori-elements-opa-writer.ts]: nit-pick: we could add a `Array.isArray()` guard to handle unexpected JSON shapes (also this scenario is currently not tested)

**Review by heimwege — DISMISSED** [review]
- changes are straightforward
- did NOT test manually
- test coverage is perfect

See one nit-pick comment but also ok if you don't do it.

File: packages/ui5-test-writer/src/fiori-elements-opa-writer.ts (hunk 2 above)
Comment (docirl) [reply packages/ui5-test-writer/src/fiori-elements-opa-writer.ts]: thanks, added

**Review by heimwege — APPROVED** [review]
re-approve

**Review by korotkovao — APPROVED** [review]
LGTM 👍 
Changest ✔️ 
Tests ✔️ 
Did not test locally
